agent/taskresource/credentialspec/credentialspec_windows.go

```diff
@@ -376,3 +456,168 @@ func (cs *CredentialSpecResource) MarshallPlatformSpecificFields(credentialSpecR
+	if err != nil {
+		seelog.Errorf("Error creating %s key: %s", taskArn, err)
+		return err
+	}
+	defer k3.Close()
+
+	err = k3.SetStringValue("AKID", taskCredentials.AccessKeyID)
+	if err != nil {
+		seelog.Errorf("Error creating AKID child value for task %s:%s", taskArn, err)
+		return err
+	}
+	err = k3.SetStringValue("SKID", taskCredentials.SecretAccessKey)
+	if err != nil {
+		seelog.Errorf("Error creating SKID child value for task %s:%s", taskArn, err)
+		return err
+	}
+	err = k3.SetStringValue("SESSIONTOKEN", taskCredentials.SessionToken)
+	if err != nil {
+		seelog.Errorf("Error creating SESSIONTOKEN child value for task %s:%s", taskArn, err)
+		return err
+	}
+
+	return nil
+}
+
+func handleNonFileDomainlessGMSACredSpec(originalCredSpec, localCredSpecFilePath, taskARN string) error {
+	// Exit early for non domainless gMSA cred specs
+	if !strings.HasPrefix(originalCredSpec, "credentialspecdomainless:") {
+		return nil
+	}
+
+	err := readWriteDomainlessCredentialSpec(localCredSpecFilePath, localCredSpecFilePath, taskARN)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func readWriteDomainlessCredentialSpec(filePath, outFilePath, taskARN string) error {
+	credSpec, err := readCredentialSpec(filePath)
+	if err != nil {
+		return err
+	}
+	err = writeCredentialSpec(credSpec, outFilePath, taskARN)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func readCredentialSpec(filePath string) (map[string]interface{}, error) {
+	byteResult, err := os.ReadFile(filePath)
+	if err != nil {
+		return nil, err
+	}
+	var credSpec map[string]interface{}
+	err = json.Unmarshal(byteResult, &credSpec)
+	if err != nil {
+		return nil, err
+	}
+	return credSpec, nil
+}
+
+func writeCredentialSpec(credSpec map[string]interface{}, outFilePath string, taskARN string) error {
+	if activeDirectoryConfigUntyped, ok := credSpec["ActiveDirectoryConfig"]; ok {
+		activeDirectoryConfig := activeDirectoryConfigUntyped.(map[string]interface{})
+		if hostAccountConfigUntyped, ok2 := activeDirectoryConfig["HostAccountConfig"]; ok2 {
+			hostAccountConfig := hostAccountConfigUntyped.(map[string]interface{})
+			if pluginInputString, ok3 := hostAccountConfig["PluginInput"]; ok3 {
+				var pluginInputParsed PluginInput
+				pluginInputString2 := pluginInputString.(string)
+				err := json.Unmarshal([]byte(pluginInputString2), &pluginInputParsed)
+				if err != nil {
+					return err
+				}
+
+				pluginInputParsed.RegKeyPath = fmt.Sprintf(RegKeyPathFormat, taskARN)
+
+				pluginInputBytes, err := json.Marshal(pluginInputParsed)
+				if err != nil {
+					return err
+				}
+
+				hostAccountConfig["PluginInput"] = string(pluginInputBytes)
+
+				jsonStr, err := json.Marshal(credSpec)
+				if err != nil {
+					return err
+				}
+
+				err = os.WriteFile(outFilePath, jsonStr, filePerm)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func getTaskIDfromARN(taskARN string) (string, error) {
```

Comment:
Can we use https://github.com/aws/amazon-ecs-agent/blob/883211e24d9255023a68f6351001c291af8a699b/agent/utils/arn.go#L16 ?